### PR TITLE
Fixes issue #170 - icon images fetched via pluginfile

### PR DESCRIPTION
**Pull request by christina-roperto**
This PR fixes issue #170 so the static images will go through pluginfile.php and can be cached for a longer period of time.

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -34,3 +34,15 @@ function block_configurable_reports_security_checks() {
...
+function block_configurable_reports_pluginfile($course, $birecord, $context, $filearea, $args, $forcedownload, $sendfileoptions) {
+    global $CFG;
+    if ($filearea !== 'export') {
+        send_file_not_found();
+    }
+    $exporttype = $args[0];
+    if (!in_array($exporttype, ['csv','json', 'ods', 'xls']) ){
+        send_file_not_found();
+    }
+    $filename = $CFG->dirroot.'/blocks/configurable_reports/'.$filearea.'/'.$exporttype.'/pix.gif';
```
Comment (brendanheywood) [inline lib.php:46]: This one doesn't feel right, shouldn't this just be normal pix serving which ends up via /theme/image.php/xxx ?

File: managereport.php [hunk 2]
```diff
@@ -164,8 +164,10 @@
         if (!empty($export)) {
             foreach ($export as $e) {
                 if ($e) {
+                    $cid = $context->id;
+                    $url = moodle_url::make_pluginfile_url($cid, 'block_configurable_reports', 'export', null, '/', $e);
                     $download .= '<a href="viewreport.php?id='.$r->id.'&amp;download=1&amp;format='.$e.'">'.
-                        '<img src="'.$CFG->wwwroot.'/blocks/configurable_reports/export/'.$e.'/pix.gif" alt="'.$e.'">'.
+                        '<img src="'.$url.'" alt="'.$e.'">'.
```
Comment (brendanheywood) [inline managereport.php:170]: please convert this to html_writer

File: report.class.php [hunk 3]
```diff
@@ -244,7 +244,10 @@ public function print_export_options($return = false) {
             $output .= get_string('downloadreport', 'block_configurable_reports').': ';
             foreach ($export as $e) {
                 if ($e) {
-                    $output .= '<a href="'.$wwwpath.'&amp;download=1&amp;format='.$e.'"><img src="'.$CFG->wwwroot.'/blocks/configurable_reports/export/'.$e.'/pix.gif" alt="'.$e.'">&nbsp;'.(strtoupper($e)).'</a>&nbsp;';
+                    $context = \context_system::instance();
+                    $cid = $context->id;
+                    $url = moodle_url::make_pluginfile_url($cid, 'block_configurable_reports', 'export', null, '/', $e);
+                    $output .= '<a href="'.$wwwpath.'&amp;download=1&amp;format='.$e.'"><img src="'.$url.'" alt="'.$e.'">&nbsp;'.(strtoupper($e)).'</a>&nbsp;';
```
Comment (brendanheywood) [inline report.class.php:250]: please convert this to html_writer

File: managereport.php (hunk 2 above)
Comment (Anupamatd) [reply managereport.php:170]: Hey @brendanheywood - this has been fixed in another pr - https://github.com/jleyva/moodle-block_configurablereports/pull/204 (i couldn't push anything to here). Please review.

Thanks,
Anu

File: report.class.php (hunk 3 above)
Comment (Anupamatd) [reply report.class.php:250]: Hey @brendanheywood - this has been fixed in another pr - https://github.com/jleyva/moodle-block_configurablereports/pull/204 (i couldn't push anything to here). Please review.

Thanks,
Anu

File: lib.php (hunk 1 above)
Comment (Anupamatd) [reply lib.php:46]: Hey @brendanheywood - I tried using theme url, but that doesn't work in pluginfile. 
In UI, the image URL is showing like it is coming from pluginfile. I hope this is fine.
image

**Pull request closed without merging**